File: app/src/main/java/io/oldering/tvfoot/red/app/domain/matches/state/MatchesService.java
package io.oldering.tvfoot.red.app.domain.matches.state;

import io.oldering.tvfoot.red.api.TvfootService;
import io.oldering.tvfoot.red.app.data.entity.Match;
import io.oldering.tvfoot.red.app.data.entity.search.Filter;
import io.reactivex.Single;
import java.util.List;
import javax.inject.Inject;

public class MatchesService {
  private final TvfootService tvfootService;
  private static final int MATCH_PER_PAGE = 30;

  @Inject MatchesService(TvfootService tvfootService) {
    this.tvfootService = tvfootService;
  }

  Single<List<Match>> loadFirstPage() {
    return tvfootService.findFuture(Filter.builder().limit(MATCH_PER_PAGE).offset(0).build());
  }

  Single<List<Match>> loadNextPage(int pageIndex) {
    return tvfootService.findFuture(
        Filter.builder().limit(MATCH_PER_PAGE).offset(MATCH_PER_PAGE * pageIndex).build());
  }
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/domain/matches/state/MatchesService.kt
package com.benoitquenaudon.tvfoot.red.app.domain.matches.state

import com.benoitquenaudon.tvfoot.red.api.TvfootService
import com.benoitquenaudon.tvfoot.red.app.data.entity.Match
import com.benoitquenaudon.tvfoot.red.app.data.entity.search.Filter
import io.reactivex.Single
import javax.inject.Inject

class MatchesService @Inject constructor(private val tvfootService: TvfootService) {
  fun loadPage(pageIndex: Int): Single<List<Match>> {
    return tvfootService.findFuture(
        Filter.builder().limit(MATCH_PER_PAGE).offset(MATCH_PER_PAGE * pageIndex).build())
  }

  companion object Constant {
    private val MATCH_PER_PAGE = 30
  }
}
